components/subscription-cta.js

```diff
@@ -14,6 +14,39 @@ function SubscriptionCTA({ description, name, prices }) {
     (price) => price.interval === activeBillingInterval
   )
 
+  const createCheckoutSession = async () => {
+    try {
+      setCheckoutLoading(true)
+
+      const checkoutSession = await fetch(
+        '/api/stripe/checkout/sessions/create',
+        {
+          method: 'POST',
+          headers: {
+            Authorization: cookie.get('first-means-everything'),
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            allow_promotion_codes: true,
+            cancel_url: window.location.href,
+            price: activePrice.id,
+            success_url: window.location.href,
+            trial_from_plan: !user.hasHadTrial
```

Comment:
Probably need to move the logic that computes this variable to the server to prevent malicious interactions.